src/Sarif/Comparers/ComparerHelpers.cs

```diff
@@ -0,0 +1,206 @@
+            {
+                result = 0;
+                return true;
+            }
+
+            if (left == null)
+            {
+                result = -1;
+                return true;
+            }
+
+            if (right == null)
+            {
+                result = 1;
+                return true;
+            }
+
+            // Cannot determine the comparison result based on reference
+            // need further comparing objects properties.
+            return false;
+        }
+
+        /// <summary>
+        /// Compare 2 lists of type T derived from IComparable.
+        /// </summary>
+        /// <typeparam name="T">type derived from IComparable</typeparam>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <returns></returns>
+        public static int CompareList<T>(IList<T> left, IList<T> right) where T : IComparable
+        {
+            return CompareListHelper(left, right, (a, b) => a.CompareTo(b));
+        }
+
+        /// <summary>
+        /// Compare 2 lists of type T using a Comparer<T>.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <param name="comparer"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static int CompareList<T>(IList<T> left, IList<T> right, IComparer<T> comparer)
+        {
+            if (comparer == null)
+            {
+                throw new ArgumentNullException(nameof(comparer));
+            }
+
+            return CompareListHelper(left, right, comparer.Compare);
+        }
+
+        /// Main function for comparing 2 lists.
+        private static int CompareListHelper<T>(IList<T> left, IList<T> right, Func<T, T, int> compareFunc)
+        {
+            if (compareFunc == null)
+            {
+                throw new ArgumentNullException(nameof(compareFunc));
+            }
+
+            if (CompareReference(left, right, out int compareResult))
+            {
+                return compareResult;
+            }
+
+            compareResult = left.Count.CompareTo(right.Count);
+
+            if (compareResult != 0)
+            {
+                return compareResult;
+            }
+
+            for (int i = 0; i < left.Count; ++i)
+            {
+                if (CompareReference(left[i], right[i], out compareResult) && compareResult != 0)
+                {
+                    return compareResult;
+                }
+
+                compareResult = compareFunc(left[i], right[i]);
+
+                if (compareResult != 0)
+                {
+                    return compareResult;
+                }
+            }
+
+            return compareResult;
+        }
+
+        public static int CompareDictionary<T>(IDictionary<string, T> left, IDictionary<string, T> right) where T : IComparable
```

Comment:
>public static int [](http://example.com/codeflow?start=8&length=17)

for all the other public methods, you added the summary. for this and a few others, you didn't add.

be consistent.

Reply:
added!